pkg/engine/networkinterfaces.go

```diff
@@ -29,13 +29,11 @@ func CreateNetworkInterfaces(cs *api.ContainerService) NetworkInterfaceARM {
 	}
 
 	lbBackendAddressPools := []network.BackendAddressPool{}
-	if !cs.Properties.OrchestratorProfile.IsPrivateCluster() {
```

Comment:
These if not conditions are not necessary as `CreateMasterVMNetworkInterfaces` is never called in a private cluster configuration flow.